src/fs/rock/RockSwapDir.cc

```diff
@@ -830,9 +833,28 @@ Rock::SwapDir::ioCompletedNotification()
            std::setw(7) << map->entryLimit() << " entries, and " <<
            std::setw(7) << map->sliceLimit() << " slots");
 
+    if (!opt_foreground_rebuild && UsingSmp() && IamDiskProcess())
+        diskerReady();
+    // else postpone until Rock::Rebuild::swanSong()
+
     rebuild();
 }
 
+void
+Rock::SwapDir::diskerReady()
+{
+    debugs(47, 7, filePath);
+    assert(IamDiskProcess());
+    assert(UsingSmp());
```

Comment:
If this can be moved into SendMessage(), please do so. They feel out of place in high-level callers that do not really rely on UsingSmp() in any way. Same for other new SendMessage() callers.